src/sort/map.rs

```diff
@@ -2,16 +2,23 @@ use super::*;
 use std::collections::BTreeMap;
 
 #[derive(Clone, Debug, PartialEq, Eq, Hash)]
-struct MapContainer<V> {
-    do_rebuild: bool,
-    data: BTreeMap<V, V>,
+pub struct MapContainer<V> {
+    do_rebuild_keys: bool,
+    do_rebuild_vals: bool,
+    pub data: BTreeMap<V, V>,
 }
 
 impl Container for MapContainer<core_relations::Value> {
     fn rebuild_contents(&mut self, rebuilder: &dyn Rebuilder) -> bool {
-        if self.do_rebuild {
+        if self.do_rebuild_keys || self.do_rebuild_vals {
+            let mut changed = false;
             let (mut keys, mut vals): (Vec<_>, Vec<_>) = self.data.iter().unzip();
```

Comment:
Should we allocate keys under `do_rebuild_keys`, `vals` under `do_rebuild_vals` etc?